src/hotspot/share/code/nmethod.cpp

```diff
@@ -1608,6 +1608,7 @@ void nmethod::post_compiled_method_load_event(JvmtiThreadState* state) {
     if (is_not_entrant() && can_convert_to_zombie()) {
       return;
     }
+    mark_as_seen_on_stack();
```

Comment:
Not obvious what this actually does in relation to the dequeuing problem.

Reply:
updated comments

Reply:
I'm not 100% this is needed because we don't handshake this thread and we just checked that it can't be converted to zombie, but I don't think this hurts anything and I thought it might be needed.